src/include/duckdb/common/types/cast_helpers.hpp

```diff
@@ -270,6 +274,11 @@ struct HugeintToStringCast {
 
 	static int DecimalLength(hugeint_t value, uint8_t width, uint8_t scale) {
 		int negative;
+
+		// Can't negate if it's the minimum value
+		if (value == NumericLimits<hugeint_t>::Minimum()) {
+			value += 1;
```

Comment:
This should never happen - decimals are bounded by their precision so a `hugeint` decimal can be at most `-999999999999999999999999999999999999999` - `999999999999999999999999999999999999999` (for a `DECIMAL(38,0)`) - so this could be an assertion I believe?